sei-db/ss/pebbledb/db.go

```diff
@@ -1085,3 +1204,130 @@ func retrieveLastRangeHashed(db *pebble.DB) (int64, error) {
 	}
 	return int64(ubz), nil
 }
+
+// collectMetricsInBackground periodically collects PebbleDB internal metrics
+func (db *Database) collectMetricsInBackground() {
+	ticker := time.NewTicker(10 * time.Second) // Collect metrics every 10 seconds
+	defer ticker.Stop()
+
+	for db.metricsCtx.Err() == nil {
+		select {
+		case <-db.metricsCtx.Done():
+			return
+		case <-ticker.C:
+			db.collectAndRecordMetrics(db.metricsCtx)
+		}
+	}
+}
+
+// collectAndRecordMetrics collects PebbleDB internal metrics and records them
+func (db *Database) collectAndRecordMetrics(ctx context.Context) {
+	if db.storage == nil {
+		return
+	}
+
+	m := db.storage.Metrics()
+
+	db.metricsMu.Lock()
+	lastSnapshot := db.lastMetricsValues
```

Comment:
Why bother with the `lastMetricsValues` struct field at all to then protect it with a lock? That field as far as I can see is only used in this function, and in this function we `Record` the metrics.

I think we can remove the mutex and `lastMetricsValues` struct field entirely.